LuaValue.h

```diff
@@ -62,12 +62,12 @@ class LuaVal
     typedef std::monostate NIL;
     template<class T> struct always_false : std::false_type {};
     typedef std::shared_ptr<MapType> WrappedMap;
-    typedef std::variant<NIL, std::unique_ptr<std::string>, WrappedMap, bool, double> LuaValVariant;
+    typedef std::variant<NIL, std::string, WrappedMap, bool, double> LuaValVariant;
```

Comment:
The original reason for using `std::unique_ptr` here was to reduce the size of a LuaVal.
The original implementation uses 24 bytes for any value, while this new version uses 40 bytes.

Reply:
In my opinion 16 bytes are worth the tradeoff of avoiding another allocation